src/platform/android/AndroidChipPlatform-JNI.cpp

```diff
@@ -274,6 +277,72 @@ JNI_LOGGING_METHOD(void, setLogFilter)(JNIEnv * env, jclass clazz, jint level)
     SetLogFilter(category);
 }
 
+static void ENFORCE_FORMAT(3, 0) logRedirectCallback(const char * module, uint8_t category, const char * msg, va_list args)
+{
+    using namespace chip::Logging;
+
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+    int priority = ANDROID_LOG_DEBUG;
+    switch (category)
+    {
+    case kLogCategory_Error:
+        priority = ANDROID_LOG_ERROR;
+        break;
+    case kLogCategory_Progress:
+        priority = ANDROID_LOG_INFO;
+        break;
+    case kLogCategory_Detail:
+        priority = ANDROID_LOG_DEBUG;
+        break;
+    default:
+        break;
+    }
+
+    jint jPriority  = static_cast<jint>(priority);
+    jstring jModule = env->NewStringUTF(module);
+    if (jModule == nullptr)
+    {
+        return;
+    }
+
+    char buffer[CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE];
+    vsnprintf(buffer, sizeof(buffer), msg, args);
+    jstring jMsg = env->NewStringUTF(buffer);
+    if (jMsg == nullptr)
+    {
+        env->DeleteLocalRef(jModule);
+        return;
+    }
+
+    env->CallVoidMethod(sJavaLogCallbackObject, sOnLogMessageMethod, jModule, jPriority, jMsg);
+
+    env->DeleteLocalRef(jModule);
```

Comment:
I have largely removed DeleteLocalRef usage in matter,  I would prefer using JniLocalReferenceScope scope(env) in the beginning of this function; then you can remove these DeleteLocalRef.
for example https://github.com/project-chip/connectedhomeip/blob/3d69583fb644577269ee468e6a28c0f4d6e40612/src/platform/android/DiagnosticDataProviderImpl.cpp#L101C28-L101C33

Reply:
Thank you, I didn't know JniLocalReferenceScope already existed. I prefer that style. 👍